Parser/asdl_c.py

```diff
@@ -1132,13 +1132,291 @@ def visitModule(self, mod):
     return result;
 }
 
+/*
+ * Perform the following validations:
+ *
+ *   - All keyword arguments are known @e fields or @e attributes.
+ *   - No field or attribute would be left unfilled after copy.replace().
+ *
+ * On success, this returns 1. Otherwise, set a TypeError
+ * exception and returns -1 (no exception is set if some
+ * other internal errors occur).
+ *
+ * @param self          The AST node instance.
+ * @param dict          The AST node instance dictionary (self.__dict__).
+ * @param fields        The list of fields (self._fields).
+ * @param attributes    The list of attributes (self._attributes).
+ * @param kwargs        Keyword arguments passed to ast_type_replace().
+ *
+ * The @e dict, @e fields, @e attributes and @e kwargs arguments can be NULL.
```

Comment:
I tend to comment everything, especially for the future me (or future someone else) but I'm not sure if you want to use doxygen-like comments (note that I did not use `/**` but only `/*` since this may render differently in IDEs and `/**` is almost never used).

If you want me to remove the doxygen directives, I can do it as well (though I think it's nice to have them).